### ISSUE TITLE
XmlWriter should allow setting not to output space before closing slash for empty elements

### ISSUE THREAD

**Issue by jez9999** [dotnet/runtime#109629]
I have an XML document I want serialized by an XmlWriter, and I'm using this code:

```
        XmlWriter xw = XmlWriter.Create(sbXmlDoc, new XmlWriterSettings {
            Indent = true,
            IndentChars = "  ",
            OmitXmlDeclaration = true
        });
```

Unfortunately, this outputs spaces before closing slashes in empty elements (`<element />` instead of `<element/>`) and I don't want the spaces there.  XmlWriter (actually I'm getting a `XmlWellFormedWriter`) should provide an option not to output these spaces, or better yet, a custom callback for doing any extra formatting you want when each element is closed.

**Comment by julealgon** [comment]
> ...and I don't want the spaces there.

@jez9999 can you elaborate a bit on why? Is this causing problems for you, if so, where?

**Comment by jez9999** [comment]
I'm diffing it with other XML documents and the serializer that generates those doesn't put the space in so it creates a bunch of annoying diffs where they're unhelpful.

**Comment by jeffhandley** [comment]
Assigned to @krwq to finish triaging this.

**Comment by krwq** [comment]
@jez9999 XmlWriter is an abstract class which means you can implement it yourself to suit your needs (i.e. you can create XmlWriter implementation which takes another XmlWriter and just call it everywhere except for WriteElement or other method where you strip the spaces as needed).

Per our [contribution bar](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.Xml/README.md#contribution-bar) we won't do this feature so I'm going to go ahead and close this issue. In case you reply with some extra info please also re-open the issue.

**Comment by jez9999** [comment]
> [@jez9999](https://github.com/jez9999) XmlWriter is an abstract class which means you can implement it yourself to suit your needs (i.e. you can create XmlWriter implementation which takes another XmlWriter and just call it everywhere except for WriteElement or other method where you strip the spaces as needed).
> 
> Per our [contribution bar](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.Xml/README.md#contribution-bar) we won't do this feature so I'm going to go ahead and close this issue. In case you reply with some extra info please also re-open the issue.

You mean like a kind of pass-through facade class?  That feels like an antipattern to me.

Why were the classes implementing `XmlWriter` made sealed so they couldn't be expanded for just this purpose?

**Comment by krwq** [comment]
Yes, I didn't make that decision, it was years before I joined the team - also they likely didn't want to make types public which have hard to describe names (XmlWriterWithSchemaValidation or whatever other super optimized versions are used underneath). As a side note XmlTextWriter is not sealed. You can do:

```csharp
using System.Text;
using System.Xml;

class MyWriter : XmlTextWriter
{
    public MyWriter(Stream s, Encoding e) : base(s, e)
    {
    }

    public override void WriteEndElement()
    {
        base.WriteEndElement();
    }
}
```

you'll likely need to also do async version if you use it and whatever else you need. The pass through is more flexible though
